controllers/cloud.redhat.com/providers/resource_cache.go

```diff
@@ -252,6 +261,19 @@ func (o *ObjectCache) Update(resourceIdent ResourceIdent, object runtime.Object)
 		return fmt.Errorf("object not found in cache, cannot update")
 	}
 
+	var gvk, obGVK schema.GroupVersionKind
```

Comment:
Since this matches the logic in `create()`, I think it would be great to factor this out to a validator function if possible.

Reply:
I actually have a deeper plan for this I'll share with you.....remind me about it next week